src/Stratis.Bitcoin/Configuration/Settings/ConnectionManagerSettings.cs

```diff
@@ -148,7 +148,7 @@ public ConnectionManagerSettings(NodeSettings nodeSettings)
             }
 
             this.BanTimeSeconds = config.GetOrDefault<int>("bantime", nodeSettings.Network.IsTest() ? DefaultMisbehavingBantimeSecondsTestnet : DefaultMisbehavingBantimeSeconds, this.logger);
-
+            this.Listen = config.GetOrDefault<bool>("listen", false, this.logger);
```

Comment:
Doesn't this default to `true` for bitcoind?

Reply:
`net.h`:

```
static const bool DEFAULT_LISTEN = true;
```

`init.cpp`:

```
    if (gArgs.IsArgSet("-bind")) {
        if (gArgs.SoftSetBoolArg("-listen", true))
            LogPrintf("%s: parameter interaction: -bind set -> setting -listen=1\n", __func__);
    }
    if (gArgs.IsArgSet("-whitebind")) {
        if (gArgs.SoftSetBoolArg("-listen", true))
            LogPrintf("%s: parameter interaction: -whitebind set -> setting -listen=1\n", __func__);
    }

    if (gArgs.IsArgSet("-connect")) {
        // when only connecting to trusted nodes, do not seed via DNS, or listen by default
        if (gArgs.SoftSetBoolArg("-dnsseed", false))
            LogPrintf("%s: parameter interaction: -connect set -> setting -dnsseed=0\n", __func__);
        if (gArgs.SoftSetBoolArg("-listen", false))
            LogPrintf("%s: parameter interaction: -connect set -> setting -listen=0\n", __func__);

...

    if (!gArgs.GetBoolArg("-listen", DEFAULT_LISTEN)) {
        // do not map ports or try to retrieve public IP when not listening (pointless)
        if (gArgs.SoftSetBoolArg("-upnp", false))
            LogPrintf("%s: parameter interaction: -listen=0 -> setting -upnp=0\n", __func__);
        if (gArgs.SoftSetBoolArg("-discover", false))
            LogPrintf("%s: parameter interaction: -listen=0 -> setting -discover=0\n", __func__);
        if (gArgs.SoftSetBoolArg("-listenonion", false))
            LogPrintf("%s: parameter interaction: -listen=0 -> setting -listenonion=0\n", __func__);
    }
```

Reply:
This happens later
```
            if (this.Listen.Count == 0)
            {
                this.Listen.Add(new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), this.Port), false));
            }
```

Reply:
@dangershony that seems to be an entirely different `Listen` though? I am referring to the default value of the bool in the settings.

In any case it would be misleading to default it to `false` if code elsewhere treats it as `true` by default.

Reply:
Ok in this case the listen is there just to override `-connect` perhaps the correct way is to explicitly set to false in case `-connect` it specified and `Listen` is not @fassadlr ?

Reply:
@zeptin thanks for the bitcoin core reference. I am going to implement it as they have done above.